### DISCUSSION TITLE
How to set a per-call timeout

### DISCUSSION THREAD

**Discussion by ncteisen** [square/wire#2338] [Q&A] [answered]
Is it possible to set a per-call timeout using wire gRPC for Kotlin?

I am looking to do something like:

```
val response = client.SomeRpcMethod().setTimeout(Seconds(5)).executeBlocking(request)
```

Is this possible? I know I could probably recreate this by using the suspend-able API, but I wondered if there was a more idiomatic way to accomplish this

**Comment by oldergod** [comment]
You can set the `timeout` field on the `GrpcCall` or `GrpcStreamingCall` object you'll get by calling the rpc method.

- https://square.github.io/wire/3.x/wire-grpc-client/com.squareup.wire/-grpc-call/
- https://square.github.io/wire/3.x/wire-grpc-client/com.squareup.wire/-grpc-streaming-call/

**Comment by ncteisen** [comment, marked as answer]
`timeout` is a val field, so it cannot be set. Do you have any code examples of overriding it?

**Reply by oldergod** [reply]
My response was confusing. The `timeout` object is mutable. Update it.

**Reply by ncteisen** [reply]
Ahh thank you I understand now. Just to double confirm (and for any future readers), a code snippet would be:

```
val grpcCall = someGrpcClient.SomeRpcMethod()
grpcCall.timeout.deadline(2, TimeUnit.SECONDS)
val response = grpcCall.execute(request)
```

**Comment by ezamelczyk** [comment]
This just adds a header to the call without actually enforcing the timeout as far as I can tell.
